src/nodes/display/FilmNode.js

```diff
@@ -0,0 +1,56 @@
+import TempNode from '../core/TempNode.js';
+import { uv } from '../accessors/UVNode.js';
+// import { luminance } from './ColorAdjustmentNode.js';
+import { addNodeElement, tslFn, nodeObject, /*vec3,*/ vec4 } from '../shadernode/ShaderNode.js';
+import { uniform } from '../core/UniformNode.js';
+import { mix, fract, clamp, rand } from '../math/MathNode.js';
+import { timerLocal } from '../utils/TimerNode.js';
+
+class FilmNode extends TempNode {
+
+	constructor( textureNode, intensity = 0.5, grayscale = false ) {
+
+		super();
+
+		this.textureNode = textureNode;
+
+		this.intensity = uniform( intensity );
+		this.grayscale = uniform( Number( grayscale ) );
+
+	}
+
+	setup() {
+
+		const { textureNode } = this;
+
+		const uvNode = textureNode.uvNode || uv();
+
+		const film = tslFn( () => {
+
+			const base = textureNode;
+
+			const noise = rand( fract( uvNode.add( timerLocal() ) ) );
+
+			let color = base.rgb.add( base.rgb.mul( clamp( noise.add( 0.1 ), 0, 1 ) ) );
+
+			color = mix( base.rgb, color, this.intensity );
+
+			// this.grayscale.equal( 1 ).cond( color.assign( vec3( luminance( color ) ) ), color ); // TOOD: equal() is always true meaning grayscale is always applied
```

Comment:
@sunag If commented out this line since the `equal()` call always evaluates to `true`. Do I use `cond()` right here?

Reply:
I think for this case we could use JS conditional

```js
if ( this.grayscale === true ) {

	color = vec3( luminance( color ) ); // assuming linear-srgb

}
```